src/briefcase/config.py

```diff
@@ -245,6 +310,12 @@ def __init__(
                 "'switch', or 'while')."
             )
 
+        for document_type, document in self.document_types.items():
```

Comment:
As a matter of terminology - `document_types` is a dictionary of `document_type_id: document_type` definitions, not `document_type: document` definitions.